Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/E2ETests/E2EAnalysisTests.cs

```diff
@@ -116,134 +128,135 @@ public void E2E_InvalidGuid()
-
-            string projectFilePath = Path.Combine(rootInputFolder,
-                descriptor.ProjectFolderName, descriptor.ProjectFileName);
-
-            projectRoot.Save(projectFilePath);
-            TestContext.AddResultFile(projectFilePath);
+<ItemGroup>
+  <Compile Include='{codeFilePath}' />
+</ItemGroup>
+";
+            var projectFilePath = CreateProjectFile(projectXml, rootOutputFolder);
 
             // Act
-            var buildLog = BuildRunner.BuildTargets(TestContext, projectFilePath);
+            var result = BuildRunner.BuildTargets(TestContext, projectFilePath);
 
             // Assert
-            buildLog.AssertTargetSucceeded(TargetConstants.DefaultBuildTarget); // Build should succeed with warnings
-            ProjectInfoAssertions.AssertProjectInfoExists(rootOutputFolder, projectFilePath);
-
-            buildLog.AssertExpectedErrorCount(0);
-            buildLog.AssertExpectedWarningCount(1);
-
-            var warning = buildLog.Warnings[0];
-            warning.Should().Contain(descriptor.FullFilePath, "Expecting the warning to contain the full path to the bad project file");
-        }
-
-        [TestMethod]
-        [TestCategory("E2E"), TestCategory("Targets")]
-        public void E2E_NoManagedFiles()
-        {
-            // Arrange
-            var rootInputFolder = TestUtils.CreateTestSpecificFolder(TestContext, "Inputs");
-            var rootOutputFolder = TestUtils.CreateTestSpecificFolder(TestContext, "Outputs");
-
-            var descriptor = BuildUtilities.CreateValidProjectDescriptor(rootInputFolder);
-
-            AddEmptyContentFile(descriptor, rootInputFolder);
-            AddEmptyContentFile(descriptor, rootInputFolder);
-            AddEmptyContentFile(descriptor, rootInputFolder);
-
-            var preImportProperties = CreateDefaultAnalysisProperties(rootInputFolder, rootOutputFolder);
+            result.AssertTargetSucceeded(TargetConstants.DefaultBuildTarget); // Build should succeed with warnings
 
-            // Act
-            var projectDir = CreateAndBuildSonarProject(descriptor, rootOutputFolder, preImportProperties);
+            var projectSpecificOutputDir = CheckProjectSpecificOutputStructure(rootOutputFolder);
+            var actualProjectInfo = CheckProjectInfoExists(projectSpecificOutputDir);
+            actualProjectInfo.ProjectGuid.Should().Be(Guid.Empty);
 
-            AssertFileExists(projectDir, ExpectedAnalysisFilesListFileName);
+            result.AssertExpectedErrorCount(0);
+            result.AssertExpectedWarningCount(1);
 
-            CheckProjectOutputFolder(descriptor, projectDir);
+            var warning = result.Warnings[0];
+            warning.Should().Contain(projectFilePath, "Expecting the warning to contain the full path to the bad project file");
         }
 
         [TestMethod]
         [TestCategory("E2E"), TestCategory("Targets")]
-        public void E2E_NoContentFiles()
+        public void E2E_HasAnalyzableFiles()
```

Comment:
Previously there were separate tests that checked whether content files and code were being analysed. I've consolidated them into this single test that contains a mixture of analyzable and non-analyzable files.